### PR TITLE
Reconcile product page header with tech docs

### PR DESCRIPTION
**Pull request by 36degrees**
We replaced the checkbox-based navigation toggle on the tech docs as although it meant Javascript wasn’t required for the toggle to work, it wasn’t as accessible as a button with the proper aria markup.

This brings the product page example in line. Unfortunately it does add a dependency on both jQuery and Modernizr (which were already required for the tech docs prior to these changes) but does mean that they are consistent in the way the navigation works.

This will hopefully allow us to extract the common styles and scripts both into a shared package of some description.

### CODE REVIEW

**Comment by 36degrees** [comment]
To test this:

- [Run the example locally](https://github.com/alphagov/product-page-example#running-example-locally) and verify that the header, including the navigation, behaves sanely – including on a mobile viewport both with and without javascript enabled.
- Verify that the header markup, javascripts/modules/navigation.js and stylesheets/modules/_header.scss are consistent between this and the tech-docs-template.

**Comment by nickcolley** [comment]
I've just had a go with VoiceOver, (not reviewed any code yet) and it doesnt seem to give a response.

Says 'menu downpointing triangle', and nothing else. Not sure if that's expected...

**Comment by 36degrees** [comment]
Are you definitely on this branch? That's not the behaviour I'm getting with Voiceover (Chrome, Mac).

I hear e.g. 'Show or hide top level navigation, collapsed, button'

File: source/javascripts/modules/navigation.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    var $nav;
+
+    this.start = function ($element) {
+      $navToggle = $('.js-nav-toggle', $element);
+      $nav = $('.js-nav', $element);
+
+      updateAriaAttributes();
+
+      $navToggle.on('click', toggleNavigation);
+      $(window).on('resize', updateAriaAttributes);
```
Comment (nickcolley) [inline source/javascripts/modules/navigation.js]: Unbounded listeners like this worry me but might be okay.

https://remysharp.com/2010/07/21/throttling-function-calls

**Pull request merged**